pyteal/ast/if_.py

```diff
@@ -26,16 +27,21 @@ def __init__(self, cond: Expr, thenBranch: Expr, elseBranch: Expr = None) -> Non
         super().__init__()
         require_type(cond.type_of(), TealType.uint64)
 
-        if elseBranch is None:
-            require_type(thenBranch.type_of(), TealType.none)
-        else:
-            require_type(thenBranch.type_of(), elseBranch.type_of())
+        if thenBranch:
```

Comment:
That makes sense. Could you also add a boolean attribute to this class indicating if the `If` was created conforming to the new `If(cond)` syntax or the existing `If(cond, ...)` syntax?

Then later when `Then`, `Else`, or `ElseIf` is called, you can raise an error if the object was not created using the new syntax. This will help avoid weird cases where a mix of the old and new syntax are used, i.e. `If(cond, then).Else(else)`.